src/api.coffee

```diff
@@ -113,12 +113,15 @@ module.exports = (application) ->
 					.finally ->
 						application.start(app)
 		.catch (err) ->
+			status = 503
+			if err instanceof utils.AppNotFoundError
+				errMsg = "App not found: an app needs to be installed for purge to work.
+					If you've recently moved this device from another app,
+					please push an app and wait for it to be installed first."
+				err = new Error(errMsg)
```

Comment:
I know it's merged, I just didn't get a chance to comment before it was :P

Ok, it works ok for now, it's just nice in general to keep stack traces intact for places where it might not be clear (I've had so many headaches trying to chase down where an error actually originates because of silenced stack traces in the past :(  )